optimum/intel/openvino/quantization.py

```diff
@@ -0,0 +1,223 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import inspect
+import io
+from itertools import chain
+from pathlib import Path
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+from datasets import Dataset, load_dataset
+from torch.onnx import export as onnx_export
+from torch.utils.data import DataLoader, RandomSampler
+from transformers import PreTrainedModel, default_data_collator
+from transformers.onnx import FeaturesManager, OnnxConfig
+
+import openvino
+import openvino.runtime.passes as passes
+from huggingface_hub import HfApi
+from nncf import NNCFConfig
+from nncf.torch import create_compressed_model, register_default_init_args
+from nncf.torch.dynamic_graph.io_handling import wrap_nncf_model_inputs_with_objwalk
+from nncf.torch.initialization import PTInitializingDataLoader
+from nncf.torch.nncf_network import NNCFNetwork
+from openvino.runtime import Core
+from optimum.quantization_base import OptimumQuantizer
+
+from .nncf_config import get_config_with_input_info
+from .utils import ONNX_WEIGHTS_NAME, OV_XML_FILE_NAME
+
+
+core = Core()
+
+
+class OVDataLoader(PTInitializingDataLoader):
+    def get_inputs(self, dataloader_output) -> Tuple[Tuple, Dict]:
+        return (), dataloader_output
+
+
+class OVQuantizer(OptimumQuantizer):
+    """
+    Handle the NNCF quantization process.
+    """
+
+    def __init__(self, model: PreTrainedModel, **kwargs):
+        """
+        Args:
+            model (`PreTrainedModel`):
+                The model to quantize.
+            seed (`int`, defaults to 42):
+                The random seed to use when shuffling the calibration dataset.
+        """
+        super().__init__()
+        self.model = model
+        self.seed = kwargs.pop("seed", 42)
+        self.feature = kwargs.pop("feature", None)
+        signature = inspect.signature(self.model.forward)
+        self._signature_columns = list(signature.parameters.keys())
+        self.input_names = None
+
+    @classmethod
+    def from_pretrained(cls, model: PreTrainedModel, **kwargs):
+        # TODO : Create model
+        return cls(model, **kwargs)
+
+    def quantize(
+        self,
+        quantization_config: NNCFConfig,
+        calibration_dataset: Dataset,
+        save_directory: Union[str, Path],
+        file_name: Optional[str] = None,
+        batch_size: int = 8,
```

Comment:
It looks reasonable to me, 8 is also the [default](https://github.com/huggingface/transformers/blob/v4.22.2/src/transformers/training_args.py#L160) value for both the training and evaluation batch size of the `TrainingArguments`, and in case of larger models, users will have the possibility to reduce it. But we can modify it if you prefer this option, I don't have a strong opinion on this

Reply:
Thanks, we can keep it for now and test memory use, and if needed change it later. I often have to manually set a much smaller batch size when training models - but I suppose a higher end system is expected there.